internal/pkg/cli/app_init.go

```diff
@@ -299,15 +314,52 @@ func (o *initAppOpts) askDockerfile() error {
 }
 
 func (o *initAppOpts) askAppPort() error {
+	// Use flag before anything else
 	if o.AppPort != 0 {
 		return nil
 	}
 
+	log.Infof(fmtParsePortFromDockerfileStart,
+		color.HighlightUserInput(o.DockerfilePath),
+		color.HighlightUserInput(o.AppName),
+	)
+
+	o.setupParser(o)
+	ports, err := o.df.GetExposedPorts()
+	if err != nil {
+		log.Debugln(err.Error())
+	}
+	var defaultPort = defaultAppPortString
+	switch len(ports) {
+	case 0:
+		log.Infof(fmtParseFromDockerfileNoPort,
+			color.HighlightUserInput(o.AppName),
+		)
+	case 1:
+		// catch the case where we had an Expose but couldn't parse a port from dockerfile
+		if err != nil {
+			log.Infof(fmtParseFromDockerfileNoPort,
+				color.HighlightUserInput(o.AppName),
+			)
+		} else {
```

Comment:
From The Go Programming Language book:
> Usually, when a function returns a non-nill error, its other results are undefined and should be ignored.

What do you think of filtering out any ports that had errors in `GetExposedPorts` itself, so that we can get rid of this if statement.

Reply:
If we get rid of this logic, we won't be able to distinguish between the case where we have only one exposed port (no error, use that port for the manifest and don't prompt) and the case where only one exposed port was parseable (error, use that port for the default and prompt). Unless there's something I misunderstand?

Reply:
I'd lean towards not having the current version as it's not obvious to a caller of `GetExposedPorts` what they should be doing with the returned data.  
If I'm a caller and I get two ports but also an error what does it mean 😛? Contrast this with always getting `nil` and a valid error. Also the returned error is only the last port that we failed to parse, what if we failed to parse two ports?

In my opinion we can keep it simple and remove the ambiguity, if we fail to parse a port in the Dockerfile then short-circuit immediately and return the error. Then the cli can prompt the user for the port.

Otherwise, we check how many ports were found.